ext/openssl/openssl.c

```diff
@@ -2032,19 +2044,28 @@ PHP_FUNCTION(openssl_x509_parse)
 	add_assoc_name_entry(return_value, "issuer", 		X509_get_issuer_name(cert), useshortnames);
 	add_assoc_long(return_value, "version", 			X509_get_version(cert));
 
-	add_assoc_string(return_value, "serialNumber", i2s_ASN1_INTEGER(NULL, X509_get_serialNumber(cert)));
+	asn1_serial = X509_get_serialNumber(cert);
 
-	/* Return the hex representation of the serial number, as defined by OpenSSL */
-	hexserial = BN_bn2hex(ASN1_INTEGER_to_BN(X509_get_serialNumber(cert), NULL));
+	bn_serial = ASN1_INTEGER_to_BN(asn1_serial, NULL);
+	/* Can return NULL on error or memory allocation failure */
+	if (!bn_serial) {
+		RETURN_FALSE;
```

Comment:
This actually also need to store error queue in 7.1+ (it means php_openssl_store_errors() that should be called every time there might be something on OpenSSL error queue to prevent crashes in other extension / libs). In practice it won't probably happen (null bn_serial) as such cert would be invalid (it means we couldn't probably get there unless I miss something) or we would have to run out of memory (malloc would fail) but better to have it there and in such case also clean the error queue. I will try to add it later...